### DISCUSSION TITLE
Infolist combine columns

### DISCUSSION THREAD

**Discussion by wpas81** [filamentphp/filament#9340] [Help] [answered]
Hi

My user models has columns for street, postcode & country. In the view user page I've built an infolist in which I want to combine these fields into one addressfield. 

I added a mutateFormDataBeforeFill function but that doesn't seem to work
Any advice?

` protected function mutateFormDataBeforeFill(array $data): array
    {
        $data['address'] = $data['street'] . PHP_EOL . $data['postcode'] . PHP_EOL . $data['country'];

        return $data;
    }`

**Comment by leandrocfe** [comment]
You can use formatStateUsing 

https://filamentphp.com/docs/3.x/infolists/entries/text/#custom-formatting

**Reply by wpas81** [reply]
thanks. I missed that one in the docs. So I got them listed in one field now but it's not adding the linebreaks
Any thoughts on that?

I kinda miss the option to create custom InfoListfields for adding some custom text (eg a description or some information) (I know I can create a custom listfield but it should be default imho)

```
Infolists\Components\TextEntry::make('street')
                    ->label('Addres')
                    ->formatStateUsing(fn (string $state, \App\Models\User $user): string => $user->street . PHP_EOL . $user->postcode . ' ' . $user->city . PHP_EOL . $user->country),
                
```

**Reply by leandrocfe** [reply, marked as answer]
I think you could pass a `->view('custom-view')` and customize the html in the blade file getting the model with `{{$getRecord()->your_field}}`

https://filamentphp.com/docs/3.x/infolists/entries/custom/#view-entries

https://filamentphp.com/docs/3.x/infolists/entries/custom/#accessing-the-eloquent-record

**Reply by wpas81** [reply]
yeah I think I'll have to do that (was hoping for a more default appraoch something liket he Placeholder in the Forms)
Thanks for your effort!!

**Reply by shomarim** [reply]
> thanks. I missed that one in the docs. So I got them listed in one field now but it's not adding the linebreaks Any thoughts on that?
> 
> I kinda miss the option to create custom InfoListfields for adding some custom text (eg a description or some information) (I know I can create a custom listfield but it should be default imho)
> 
> ```
> Infolists\Components\TextEntry::make('street')
>                     ->label('Addres')
>                     ->formatStateUsing(fn (string $state, \App\Models\User $user): string => $user->street . PHP_EOL . $user->postcode . ' ' . $user->city . PHP_EOL . $user->country),
>                 
> ```

Add `->html()` to your `TextEntry` and change PHP_EOL to `<br/>`. This will add the line breaks.
